### ISSUE TITLE
[FEA] cudf parquet reader chokes when more than one entry

### ISSUE THREAD

**Issue by lmeyerov** [rapidsai/cudf#6852]
**Describe the bug**

When `to_parquet` generates a folder with multiple `my.parquet/part.<i>.parquet` files, `cudf.read_parquet('my.parquet')` fails with `FileNotFoundError`. In contrast, `pd.read_parquet` / `dask_cudf.read_parquet` still work. 

**Steps/Code to reproduce bug**

```
%%time
! rm -rf lines10M*
! ( echo "Line,F2,F3,N,I" \
    && perl -E 'for($i=0;$i<10000000;$i++){say "Line $i,field2,field3,",int rand 100,",$i" }' \
  ) > lines10M.csv
! echo "generated csv, now making parquet" && ls -alh lines10M.csv
dask_cudf.read_csv('lines10M.csv').to_parquet('lines10M.parquet', compression='snappy')
! du -hs lines10M.parquet

%%time
len(cudf.read_parquet('lines10M.parquet', columns=['F2', 'N']))
```

=>

```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<timed eval> in <module>

/conda/envs/rapids/lib/python3.7/site-packages/cudf/io/parquet.py in read_parquet(filepath_or_buffer, engine, columns, filters, row_groups, skiprows, num_rows, strings_to_categorical, use_pandas_metadata, *args, **kwargs)
    248             num_rows=num_rows,
    249             strings_to_categorical=strings_to_categorical,
--> 250             use_pandas_metadata=use_pandas_metadata,
    251         )
    252     else:

cudf/_lib/parquet.pyx in cudf._lib.parquet.read_parquet()

cudf/_lib/parquet.pyx in cudf._lib.parquet.read_parquet()

cudf/_lib/io/utils.pyx in cudf._lib.io.utils.make_source_info()

FileNotFoundError: [Errno 2] No such file or directory: ['/home/graphistry/notebooks/tests/lines10M.parquet']
```

**Expected behavior**

File gets read back.  

* dask_cudf + pd readers succeed
* cudf succeeds when reduced to 10K rows (1 file, vs multi-part folder)

**Environment overview (please complete the following information)**

rapids 0.16 (latest stable) via conda to docker

**Comment by vuule** [comment]
This is more of a feature request, currently `cudf.read_parquet` expects file names of each input file, not the folder names.
There is a similar feature being added: https://github.com/rapidsai/cudf/issues/5683 but it does not address this request.

**Comment by vuule** [comment]
@lmeyerov is this issue blocking you? 
AFAICT the workaround is to call with the list of files in the directory instead, but I'm not sure if this works for you.

**Comment by lmeyerov** [comment]
I appreciate you asking @vuule  -- this kind of thing we're fine working around, just thought folks show know

**Comment by kkraus14** [comment]
cc @ayushdg we may be able to handle this in #6815 relatively easily

**Comment by ayushdg** [comment]
> cc @ayushdg we may be able to handle this in #6815 relatively easily

Thanks for the ping! Once #6815 lands, it should be possible to read the files with a `/path/folder.parquet/*.parquet` call. Reading a parquet folder directly might not be possible yet, since the folder might also contain additional metadata or success files which would need to be filtered out.

**Comment by vuule** [comment]
@ayushdg can we implement directory path support by detecting that input path is a dir and replacing the path with path/*.parquet? The #6815 takes care of the rest.

**Comment by ayushdg** [comment]
> @ayushdg can we implement directory path support by detecting that input path is a dir and replacing the path with path/*.parquet? The #6815 takes care of the rest.

Looking and pandas/dask, they pass the directory path to PyArrow and allows PyArrow to return the relevant parquet files. I think for now replacing `path` with `path/*.parquet` seems like an effective solution that I could add to #6815. Thanks!

**Comment by ayushdg** [comment]
Should be resolved by #6815. 
Note: This will work for directories containing multipart parquet files but _not_ work for parquet datasets that are partitioned by columns. Feel free to reopen if you encounter issues!

**Comment by lmeyerov** [comment]
Thanks! 

And yeah -- the main property I was testing here was ensuring `read_parquet(write_parquet(df))` works in the default case. The repro added non-default `compression='snappy'`, but I believe that's supposed to be the default and is being caught elsewhere. Hopefully the column restriction won't come up w/ default write settings..
